sycl/source/device_selector.cpp

```diff
@@ -137,10 +153,20 @@ select_device(const DSelectorInvocableType &DeviceSelectorInvocable,
 /// 2. CPU
 /// 3. Host
 /// 4. Accelerator
+
+static void traceDeviceSelector(const std::string &DeviceType) {
+  bool ShouldTrace = false;
+  ShouldTrace = detail::pi::trace(detail::pi::TraceLevel::PI_TRACE_BASIC);
+  if (ShouldTrace) {
+    std::cout << "SYCL_PI_TRACE[all]: Requested device_type: " << DeviceType << std::endl;
+  }
+}
+
 __SYCL_EXPORT int default_selector_v(const device &dev) {
   // The default selector doesn't reject any devices.
   int Score = 0;
 
+  traceDeviceSelector("default");
```

Comment:
Shouldn't this be "info::device_type::automatic"?

Reply:
Agree, updated.